integration-test/src/test/java/org/sagebionetworks/IT049FileHandleTest.java

```diff
@@ -558,6 +558,39 @@ public void testMultipartUploadV2Reset() throws FileNotFoundException, SynapseEx
 		assertFalse(startStatus.getUploadId().equals(statusAgain.getUploadId()));
 	}
 
+	@Test
+	public void testCreateExternalS3FileHandleFromExistingFile() throws Exception {
+		assertNotNull(imageFile);
+		assertTrue(imageFile.exists());
+		String md5 = MD5ChecksumHelper.getMD5Checksum(imageFile);
+
+		// Upload the owner.txt to S3 so we can create the external storage location
+		String baseKey = "integration-test/IT049FileHandleTest/testCreateExternalS3FileHandleFromExistingFile/" + UUID.randomUUID().toString();
+		String key = baseKey + FILE_NAME;
+		uploadOwnerTxtToS3(config.getS3Bucket(), baseKey, synapse.getUserProfile(userToDelete.toString()).getUserName());
+
+		// upload the little image to S3, but not through Synapse
+		synapseS3Client.putObject(config.getS3Bucket(), key, imageFile);
+

```

Comment:
The object does not necessarily exist in S3 at the time the client returns from the putObject() call.  See other cases in which putObject() is called to see how we wait for the object to appear before continuing.